### DISCUSSION TITLE
Return different output based on input variable

### DISCUSSION THREAD

**Discussion by nitse** [microsoft/RulesEngine#452] [Q&A]
I am trying to write a rule that returns a different output depending on the value of the input. 
I have an age range  and I want to return different "points" based on user's age. All the examples that I have found have a standard output.
I want to achieve something like the following while avoiding to  create a different rule for every age range.:

age              output
<55       =     +0 
55-59    =     +1 
60-64    =     +2
etc
 
Is it possible?
Thanks !!

**Comment by bmasotta** [comment]
As a RulesEngine user, I see the following:

1. a rule just evaluate to true or false (Success or Failure)
2. a rule always expose an ActionResult in its RuleResultTree,
3. if there's no applicable CustomAction defined, the ActionResult will have .Ouput = to rule's RuleResultTree
4. if you define a normal CustomAction that will be like writing static c# code, that's not what you want
5. BUT... there are two special (built in) kind of actions added silently and automatically: the OutputExpression and the EvaluateRule.
    you can use the first one to do what you need, but... at the end you will be creating a rule inside the rule actions

maybe it's better to build several rules matching rule ranges by using an Operator = "OrElse", each rule having the range of interest, but **having an OutputExpression CustomAction that returns a Global or Local Param**

i.e. (not checked):

GlobalParams (or local, inside the corresponding rule)
  AgeRankIrrelevant: someValue
  AgeRankConsumer: otherValue
Rules:
  Operator: "OrElse"
  Rule1: 
    Expression: "input1.Age  <= 10"
    CustomActions: 
      Name: "OutputExpression"
      Expression: "AgeRankIrrelevant"
  Rule2: 
    Expression: "input1.Age  > 10 && input1.age <= 20"
    CustomActions: 
      Name: "OutputExpression"
      Expression: "AgeRankConsumer"
